### PR TITLE
Homework 1

### PR DESCRIPTION
**Pull request by kirhgoff**
for all numbers below 10 (excluding 0)
result of multiplication of all even numbers is 384=2*4*6*8
calculate the multiplication result of all even numbers below 100000

### CODE REVIEW

File: multiples.py [hunk 1]
```diff
@@ -15,6 +16,16 @@ def good(i):
     return by_3 or by_5
 
 
+def even_elements_mult(i):
+    if i == 0:
+        return False
+    for num in i:
+        if num % 2 == 0 and num > 0:
+            print (num)
+        result = num * num
```
Comment (kirhgoff) [inline multiples.py]: There are multiple problems:
1) you define result inside the `for` block (I am surprised it does work, actually) - compare it with the sum in the previous example - we define it at the beginning and add TO IT  all the way. Here for every for loop iteration, you define a variable named `result` and assign it to num ^ 2. So when you finish you are returning the last value of the result which was 9*9 (last number in the loop)
2) Identation. Line `result = num * num` is not in the if block,  it is on the same level, meaning you will not filter odd numbers and you will call this line for EVERY number in the loop
3) I am not sure if this is valid syntax for the loop `for num in another_num:`, surprised this does work, you are supposed to use range(i) or at least explain what is that syntax you are using.

Comment (irinaBaB) [reply multiples.py]: please have a look on the latest changes

Comment (irinaBaB) [reply multiples.py]: https://github.com/irinaBaB/python_exercise/commit/4c64bc2ad1e43ff3386b24d8b1206d739c32fc24

File: tests/test_evenNumbers.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import array as arr
+from multiples import *
+
+
+class EvenNumbers(unittest.TestCase):
+
+    def test_multiply_even_numbers(self):
+        x = [i for i in range(10)]
+        multiplication_even=even_elements_mult(x)
+        self.assertEqual(384, multiplication_even)
```
Comment (kirhgoff) [inline tests/test_evenNumbers.py:11]: Need a newline at the end of the file? Wonder why didnt PEP tell you that

File: multiples.py [hunk 3]
```diff
@@ -15,6 +16,14 @@ def good(i):
     return by_3 or by_5
 
 
+def even_elements_mult(i):
```
Comment (kirhgoff) [inline multiples.py:19]: Bad name for the variable, the name should explain the nature of what is it, I suggest something like `limit`

File: multiples.py [hunk 4]
```diff
@@ -15,6 +16,14 @@ def good(i):
...
+    result = 1
+    for num in i:
```
Comment (kirhgoff) [inline multiples.py:21]: Are you sure you can use for loops like that? Is it a valid way to define a for loop?

File: multiples.py [hunk 5]
```diff
@@ -15,6 +16,14 @@ def good(i):
     return by_3 or by_5
```
Comment (kirhgoff) [inline multiples.py:16]: I suggest you simplify this code as well. Challenge: make it all fit on one line

File: tests/test_evenNumbers.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import unittest
```
Comment (kirhgoff) [inline tests/test_evenNumbers.py:1]: Filename is incorrect: it should be snake case - test_even_numbers.py

File: multiples.py [hunk 7]
```diff
@@ -15,6 +16,14 @@ def good(i):
...
 
 
+def even_elements_mult(i):
+    result = 1
+    for num in i:
+        if num % 2 == 0 and num > 0:
```
Comment (kirhgoff) [inline multiples.py:22]: You will be able to remove second condition `num > 0` if you correctly create for loop using range. How?

**Review by kirhgoff — COMMENTED** [review]
pylint is not happy with the code:
```
→ pylint *.py                                                                                                                                      [6be3c66]
************* Module multiples
multiples.py:24:18: C0326: No space allowed before bracket
            print (num)
                  ^ (bad-whitespace)
multiples.py:36:0: C0304: Final newline missing (missing-final-newline)
multiples.py:1:0: C0111: Missing module docstring (missing-docstring)
multiples.py:3:0: C0111: Missing function docstring (missing-docstring)
multiples.py:11:0: C0111: Missing function docstring (missing-docstring)
multiples.py:19:0: C0111: Missing function docstring (missing-docstring)
multiples.py:1:0: W0611: Unused array imported as arr (unused-import)

-----------------------------------
Your code has been rated at 6.67/10
```

File: multiples.py (hunk 4 above)
Comment (irinaBaB) [reply multiples.py:21]: yes

Comment (irinaBaB) [reply multiples.py:21]: I googled it

**Comment by irina-babicheva-myob-com** [comment]
good

**Pull request merged**